crates/bevy_pbr/src/atmosphere/render_sky.wgsl

```diff
@@ -32,31 +34,56 @@ struct RenderSkyOutput {
 @fragment
 fn main(in: FullscreenVertexOutput) -> RenderSkyOutput {
     let depth = textureLoad(depth_texture, vec2<i32>(in.position.xy), 0);
-
-    let ray_dir_ws = uv_to_ray_direction(in.uv);
+    let world_pos = get_view_position();
     let r = view_radius();
-    let mu = ray_dir_ws.y;
+    let ray_dir_ws = uv_to_ray_direction(in.uv).xyz;
+    let up = normalize(world_pos);
+    let mu = dot(ray_dir_ws, up);
+
+    let should_raymarch = settings.rendering_method == 1u;
+
+    let raymarch_steps = 16.0;
 
     var transmittance: vec3<f32>;
     var inscattering: vec3<f32>;
 
     let sun_radiance = sample_sun_radiance(ray_dir_ws.xyz);
 
     if depth == 0.0 {
-        let ray_dir_as = direction_world_to_atmosphere(ray_dir_ws.xyz);
+        let ray_dir_as = direction_world_to_atmosphere(ray_dir_ws);
         transmittance = sample_transmittance_lut(r, mu);
         inscattering += sample_sky_view_lut(r, ray_dir_as);
-        inscattering += sun_radiance * transmittance * view.exposure;
+        if should_raymarch {
+            let t_max = max_atmosphere_distance(r, mu);
+            let sample_count = mix(1.0, raymarch_steps, clamp(t_max * 0.01, 0.0, 1.0));
+            let result = raymarch_atmosphere(world_pos, ray_dir_ws, t_max, sample_count, in.uv, true, true, true);
+            inscattering = result.inscattering;
+            transmittance = result.transmittance;
+        }
+
+        inscattering += sun_radiance * transmittance;
     } else {
         let t = ndc_to_camera_dist(vec3(uv_to_ndc(in.uv), depth));
         inscattering = sample_aerial_view_lut(in.uv, t);
         transmittance = sample_transmittance_lut_segment(r, mu, t);
+
+        if should_raymarch {
+            let sample_count = mix(1.0, raymarch_steps, clamp(t * 0.01, 0.0, 1.0));
+            let result = raymarch_atmosphere(world_pos, ray_dir_ws, t, sample_count, in.uv, true, false, true);
+            inscattering = result.inscattering;
+            transmittance = result.transmittance;
+        }
     }
+
+    // exposure compensation
+    inscattering *= view.exposure;
```

Comment:
is this physically based or is it a "look good" factor? what is it compensating for

Reply:
Previously the exposure multiplication was inside of the functions, see other comment